### PR TITLE
Fixed binary classes symbol presentation flaws

### LINKED ISSUE #165: Math expressions involving summation symbol show flaws in presentation
*Transferred from https://github.com/metanorma/metanorma-standoc/issues/796*
*In relation to https://github.com/metanorma/pdfa-iso-32000-2/issues/5*

Markup sample:
```adoc
S(u,v) = sum_{i=0}^3 sum_{j=0}^3 p_{ij} xx B_i(u) xx B_j (v)
````

DOC output:
issue15-generated-doc

Original:
issue15-original

### PR DESCRIPTION
**Pull request by suleman-uzair**
This PR fixes summation symbols flaws in the presentation by upgrading 'sum', 'prod', and 'int' to ternary classes and updating OMML conversion methods.
This PR also updates.
1. Parser rules for all the currently supported markup languages.
2. Transformation rules for all the currently supported markup languages.
3. Rspecs of all the currently supported markup languages.

closes #165

### CODE REVIEW

File: lib/plurimath/math/function/ternary_function.rb [hunk 1]
```diff
@@ -65,15 +67,35 @@ def class_name
...
+
+        def validate_function_formula(field)
+          unary_classes = (Utility::UNARY_CLASSES + ["obrace", "ubrace", "hat"])
+          if field.is_a?(Formula)
+            !(field.value.any?(Left) || field.value.any?(Right))
+          elsif unary_classes.include?(field.class_name)
+            false
+          elsif field.class.name.include?("Function") || field.is_a?(FontStyle)
+            !field.is_a?(Text)
+          end
```
Comment (HassanAkbar) [inline lib/plurimath/math/function/ternary_function.rb:86]: This should be moved to respective classes.

File: lib/plurimath/math/function/ternary_function.rb [hunk 2]
```diff
@@ -112,6 +134,80 @@ def latex_wrap(field)
...
+            Array(parameter_value)
+          end
+        end
+
+        def underover_class?(field)
+          if field.is_a?(Symbol)
+            ["&#x22c0;", "&#x22c1;", "&#x22c2;", "&#x22c3;"].include?(field.value)
+          else
+            ["ubrace", "obrace"].include?(field.class_name)
+          end
```
Comment (HassanAkbar) [inline lib/plurimath/math/function/ternary_function.rb:205]: I think wherever we are using `if .is_a?(Classname)` type check can be moved to methods in the respective class. 
That way here we can simply do something like `field.underover_class?` here.

**Review by HassanAkbar — CHANGES_REQUESTED** [review]

**Review by ronaldtse — APPROVED** [review]

**Comment by ronaldtse** [comment]
Thank you @suleman-uzair , can you please help release soon?

**Comment by suleman-uzair** [comment]
> Thank you @suleman-uzair , can you please help release soon?

Version 0.3.6 is released.

**Pull request merged**